src/Sylius/Bundle/ResourceBundle/Storage/CookieStorage.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\ResourceBundle\Storage;
+
+use Sylius\Component\Resource\Storage\StorageInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\Cookie;
+use Symfony\Component\HttpFoundation\ParameterBag;
+use Symfony\Component\HttpKernel\Event\FilterResponseEvent;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+/**
+ * @author Kamil Kokot <kamil.kokot@lakion.com>
+ */
+final class CookieStorage implements StorageInterface, EventSubscriberInterface
```

Comment:
As a reviewer, I have provided you a logic reasons, why do I think it should be placed in the sperate class. It is not my duty to write it for you. You can agree with them or not. I am also not responsible for merging it or not.
Saying, that this class has _one_ responsibility is like putting all the code into the controller and saying, that this class has also only _one_ responsibility: generate the response. This class updates a storage on each request as well as providing a cookie storage behaviour (I agree, that it is not a typical one).
Anyway, I don't care about the color of your bikeshed. It is working. If it will be merged I'm ok with that. But for me, storage should be a separate class.